nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -645,8 +645,99 @@ impl StringPrototype {
         ))
     }
 
-    fn replace(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// ### [22.1.3.19 String.prototype.replace ( searchValue, replaceValue )](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.replace)
+    fn replace(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+
+        let search_value = args.get(0);
+        let replace_value = args.get(1);
+
+        // Early path: use rust's replace if everything are strings: `"foo".replace("o", "a")`
+        if let (Ok(s), Ok(search_string), Ok(replace_string)) = (
+            String::try_from(o),
+            String::try_from(search_value),
+            String::try_from(replace_value),
+        ) {
+            let result = s
+                .as_str(agent)
+                .replace(search_string.as_str(agent), replace_string.as_str(agent));
+            return Ok(String::from_string(agent, result).into_value());
+        }
+
+        // 2. If searchValue is neither undefined nor null, then
+        if !search_value.is_null() && !search_value.is_undefined() {
```

Comment:
**issue:** We can't actually move the fast path to be before this step :( 

```
String.prototype[Symbol.replace] = () => "test";
"foobar".replace("") // "test"
```

Though optionally we can do this check on the fast-path with a `get_method` operating on `agent.current_realm().intrinsics().string_prototype()` instead of on `search_string`. That's still a somewhat valuable skip :)